### DISCUSSION TITLE
Please run mitmproxy in an interactive shell environment

### DISCUSSION THREAD

**Discussion by thethreesolarian** [mitmproxy/mitmproxy#4970] [Q&A]
Hey guys, I hope you all are doing well.

I am in big trouble here. I need to youse your software to collect the HTTP response of a CURL requests on our server. I integrated the mitmproxy to be automatically installed via Ansible. So far so good. But when Ansible reaches the moment where I start the mitmproxy, the Jenkins spit out this error and fails to deploy:

 _console interface requires a tty. Please run mitmproxy in an interactive shell environment_

Please, I need to use this and not the mitmdump. Is there any way to make it work in this interactive mode?

Thank you and stay safe all.

**Comment by mhils** [comment]
What do you need mitmproxy for compared to mitmdump? Your shell environment seems to be non-interactive, so you likely couldn't interact with it.

**Reply by thethreesolarian** [reply]
I needed it to log the full body response of a CURL requests which I am not able to do with mitmdump

**Reply by mhils** [reply]
Make sure that you spawn mitmproxy is a proper TTY environment. `sys.stdout.isatty()` needs to return `True` in your setup.
